src/Chapter1.hs

```diff
@@ -554,7 +556,14 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | x' <= y' && x' <= z' = if y' < z' then y else z
+    | y' <= x' && y' <= z' = if x' < z' then x else z
+    | z' <= x' && z' <= x' = if y' < x' then y else x
```

Comment:
I think that there is no need to check abs values here, as you need to find the middle not by the abs value 🙂 

Also, if you change the check a bit, you can avoid the `if` in each case, like when you check that the element in the middle:

```haskell
| x <= y && y <= z = y
...
```

As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = ...`, to tell the compiler, that your pattern matching is exhaustive 🙂
However, as you correctly notice, your guards cover all possible cases. That means that if the pattern would not match first two, then it will satisfy the last one for sure. So you can skip the checks, and say that `otherwise = z` :)
Hope it makes sense!